imap/imap_proxy.c

```diff
@@ -473,16 +473,35 @@ void print_listresponse(unsigned cmd, const char *extname, char hier_sep,
 
     prot_printastring(imapd_out, extname);
 
-    if (attributes & MBOX_ATTRIBUTE_CHILDINFO_SUBSCRIBED) {
-        prot_puts(imapd_out, " (CHILDINFO (");
-        /* RFC 5258:
-         *     ; Note 2: The selection options are always returned
-         *     ; quoted, unlike their specification in
-         *     ; the extended LIST command.
-         */
-        if (attributes & MBOX_ATTRIBUTE_CHILDINFO_SUBSCRIBED)
-            prot_puts(imapd_out, "\"SUBSCRIBED\"");
-        prot_puts(imapd_out, "))");
+    if (oldname || (attributes & MBOX_ATTRIBUTE_CHILDINFO_MASK)) {
+        sep = "";
+        prot_puts(imapd_out, " (");
+
+        if (oldname) {
```

Comment:
@brong OLDNAME is also required for NOTIFY (RFC 5465) which is why I left it ungated.  I don't believe that we would ever set oldname when neither IMAP4rev2 nor NOTIFY are active, but I can add a check if we feels it could be a problem.